extensions/VisualEditor/modules/ve-mw/init/targets/ve.init.mw.ViewPageTarget.js

```diff
@@ -587,12 +607,24 @@ ve.init.mw.ViewPageTarget.prototype.onSaveErrorCaptcha = function () {
 			'callback': function ( response ) {
 				this.captchaResponse = response;
 			}.bind( this )
-		}
-	);
-	this.saveDialog.$frame.addClass( 'oo-ui-window-frame-captcha' );
-	this.saveDialog.popPending();
+		});
+};
 
-	this.events.trackSaveError( 'captcha' );
+/**
+ * If reCaptcha failed to load (eg, the user is in China and google is blocked),
+ * load Fancy Captcha and render it instead.
+ *
+ * @method
+ */
+ve.init.mw.ViewPageTarget.prototype.loadAndRenderFancyCaptcha = function () {
+	$.nirvana.sendRequest({
+		controller: 'CaptchaController',
+		method: 'getFancyCaptcha',
+		type: 'GET',
+		callback: function (data) {
```

Comment:
VE actually follows the MW whitespace guidelines, so there should be spaces in all these parentheses.